doc/specs/#5000 - Process Model 2.0/#5000 - Process Model 2.0.md

```diff
@@ -0,0 +1,1204 @@
+the other process could be killed. At any point, a content process could be
+killed, without the window being closed. The window will need to be redundant to
+such a scenario, and if the content process is killed, display an appropriate
+error message, but _continue running_.
+
+When the monarch process dies, we'll need to be able to reliably elect a new
+monarch. While we're electing a new monarch, and updating the new monarch,
+there's always the chance that the _new_ monarch is also killed (This is the
+"Pope Stephen II" scenario). In this scenario, if at any point a peasant
+notices that the monarch has died, the peasant will need to begin checking who
+the new monarch is again.
+
+There is certain to be a long tail of edge cases we'll discover as a product of
+this change. We'll need to be prepared to see an inevitable decrease in initial
+reliability numbers, and increased bug reports for a time. Additionally, these
+bugs will likely be fairly difficult to track down. Initially, it may help us to
+track these bugs down if we add additional tracing around each of the
+inter-process calls we might make. Some example situations might include:
+* When a process creates a new content process
+* When a content process is attached by a window
+* When a monarch is elected
+* when a peasant receives its new ID from a monarch
+* when either a window or content process _safely_ exits
+
+In any and all of these situations, we'll want to try and be as verbose as
+possible in the logging, to try and make tracking which process had the error
+occur easier.
+
+In the long run, this may end up _increasing_ reliability, as crashes in the
+buffer should no longer cause the _entire_ terminal application to crash.
+
+
+</td>
+</tr>
+<tr>
+<td><strong>Compatibility</strong></td>
+<td>
+
+The biggest concern regarding compatibility will be ensuring that the Universal
+app version of the Windows Terminal will still work as before. Although that
+application isn't used by any customer-facing scenarios currently, it still
+represents some long term goals for the Terminal.  We'll probably need to
+disable tear-out _entirely_ on the universal app, at least to begin with.
+Additionally, we'll need to ensure that all the controls in the universal
+application are in-proc controls, not using the window/content split process
+model.
+
+I'm also concerned that each individual atomic step along the path towards this
+goal still works. For more on this topic, see [Implementation
+Plan](#implementation-plan).
+
+</td>
+</tr>
+<tr>
+<td><strong>Performance, Power, and Efficiency</strong></td>
+<td>
+
+There's no dramatic change expected here. There may be a minor delay in the
+spawning of new terminal instances, due to requiring cross-process hops for the
+instantiation of the content process.
+
+Additionally, minor delays are expected to be introduced during startup for the
+initial setup of the monarch/peasant relationship.
+
+</td>
+</tr>
+</table>
+
+## Potential Issues

```

Comment:
Note to self: I stopped reading here. Some of these scenarios may come up in the spec meeting. I'll read through this after that.